### DISCUSSION TITLE
Copy the Java SDK SecretCache

### DISCUSSION THREAD

**Discussion by NikolayMetchev** [aws/aws-sdk-kotlin#1286] [Ideas]
the Java SDK has a class called SecretCache. https://docs.aws.amazon.com/secretsmanager/latest/userguide/retrieving-secrets_cache-java-ref_SecretCache.html
Can we add that to the Kotlin SDK. Or should we use the Java version directly?

**Comment by lauzadis** [comment]
That SecretCache is a custom library handwritten for the Java SDK. https://github.com/aws/aws-secretsmanager-caching-java

We currently don't plan to implement a custom handwritten secrets caching library, but I'll leave this [open as a feature request](https://github.com/awslabs/aws-sdk-kotlin/issues/1288) for other to provide feedback and help us prioritize the issue.
